Add Wasmtime latest/LTS support with LTS default

jsturtevant opened the pull request:
Introduces wasmtime_lts feature flag to switch between wasmtime versions.
Default uses latest (45), pass --lts flag to AOT compiler for LTS version (36.). CI updated to test both.

ludfjig commented on src/wasm_runtime/src/main.rs:
```diff
@@ -19,6 +19,20 @@ limitations under the License.
+extern crate wasmtime;
+#[cfg(feature = "wasmtime_lts")]
+extern crate wasmtime_lts as wasmtime;
+
```
Is there any value in making these not mutually exclusive? If both are enabled, we could default to one of them, similar to mshv and mshv3 used to work in hyperlight. I think mutually exclusive features are generally discouraged

jsturtevant committed:
Add selectable Wasmtime versions

Default runtime builds to the Wasmtime LTS dependency while keeping latest Wasmtime available via an explicit feature and AOT CLI selection.

Update runtime compatibility, embedded Wasmtime metadata, AOT precompilation, local recipes, and documentation for the selected version.

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
Update CI for Wasmtime version matrix

Run PR and release workflows against the LTS default and a reduced latest Wasmtime matrix, and pass computed feature sets consistently through the Rust workflow.

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
Update to latest wasmtime versions

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

simongdavies commented on .github/workflows/dep_rust.yml:
```diff
@@ -40,6 +40,15 @@ jobs:
           hypervisor: [hyperv, mshv3, kvm] # hyperv is windows, mshv and kvm are linux
           cpu: [amd, intel]
           config: [debug, release]
+          wasmtime: [latest, lts]
```
Do these skipped ones get tested anywhere? should we maybe randomise what we skip?

jsturtevant replied:
I am skipping the debug builds, I could add a nightly or something but I guess i was less worried about the debug builds

jsturtevant replied:
oh woops maybe i am skipping more than I thought

jsturtevant replied:
updated, I must have messed it up when rebasing...

jsturtevant committed:
simplify the build flags

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
Make note of post-return

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
Don't turn this on with pully

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
Make the selection of the version more concrete

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
fix dropped function_call_metrics_test

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>

jsturtevant committed:
only skip latest+debug

Signed-off-by: James Sturtevant <jsturtevant@gmail.com>
